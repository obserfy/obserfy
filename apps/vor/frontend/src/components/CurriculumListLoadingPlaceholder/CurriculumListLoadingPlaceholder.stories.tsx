import { Meta, Story } from "@storybook/react"
import CurriculumListLoadingPlaceholder from "./CurriculumListLoadingPlaceholder"

export default {
  title: "Core/CurriculumListLoadingPlaceholder",
  component: CurriculumListLoadingPlaceholder,
  parameters: {
    componentSubtitle: "Just a simple CurriculumListLoadingPlaceholder",
  },
} as Meta

const Template: Story = (args) => <CurriculumListLoadingPlaceholder {...args} />

export const Default = Template.bind({})
Default.args = {}
