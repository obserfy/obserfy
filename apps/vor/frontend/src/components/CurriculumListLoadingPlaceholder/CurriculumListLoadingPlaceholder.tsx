import { FC } from "react"
import { Box } from "theme-ui"
import { borderBottom } from "../../border"
import { range } from "../../domain/array"
import LoadingPlaceholder from "../LoadingPlaceholder/LoadingPlaceholder"

const CurriculumListLoadingPlaceholder: FC<{
  length?: number
}> = ({ length = 2 }) => (
  <>
    {range(length).map(() => (
      <>
        <Box sx={{ ...borderBottom }}>
          <LoadingPlaceholder m={3} sx={{ height: 21, width: 100 }} />
        </Box>
        <Box sx={{ ...borderBottom }}>
          <LoadingPlaceholder m={3} sx={{ height: 21, width: 200 }} />
        </Box>
        <Box sx={{ ...borderBottom }}>
          <LoadingPlaceholder m={3} sx={{ height: 21, width: 150 }} />
        </Box>
      </>
    ))}
  </>
)

export default CurriculumListLoadingPlaceholder
