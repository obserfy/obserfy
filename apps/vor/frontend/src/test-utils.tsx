/* eslint-disable import/no-extraneous-dependencies */
import { FC, ReactElement } from "react"
import { QueryClient, QueryClientProvider } from "react-query"
import { ThemeProvider, Box, Theme } from "theme-ui"
import { render, RenderOptions, RenderResult } from "@testing-library/react"
import {
  createHistory,
  createMemorySource,
  LocationProvider,
} from "@reach/router"
import { I18nProvider } from "@lingui/react"
import { i18n } from "@lingui/core"
import { en } from "make-plural/plurals"
import theme from "./gatsby-plugin-theme-ui"
// @ts-ignore
import enCatalog from "../i18n/lingui/en/messages.js"

// any used here because the catalogs are generated b lingui
// not our responsibility
i18n.loadLocaleData("en", { plurals: en })
i18n.load("en", enCatalog.messages as any)
i18n.activate("en")

const queryClient = new QueryClient()
const history = createHistory(createMemorySource("/"))

const AllProviders: FC = ({ children }) => {
  return (
    <I18nProvider i18n={i18n}>
      <QueryClientProvider client={queryClient}>
        <LocationProvider history={history}>
          <ThemeProvider theme={theme as Theme}>
            <Box sx={{ fontSize: [16, 20] }}>{children}</Box>
          </ThemeProvider>
        </LocationProvider>
      </QueryClientProvider>
    </I18nProvider>
  )
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "queries">
): RenderResult => render(ui, { wrapper: AllProviders, ...options })

// re-export everything
export * from "@testing-library/react"

// override render method
export { customRender as render }
