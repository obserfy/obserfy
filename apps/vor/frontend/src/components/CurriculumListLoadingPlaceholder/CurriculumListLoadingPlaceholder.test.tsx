import { render } from "../../test-utils"
import CurriculumListLoadingPlaceholder from "./CurriculumListLoadingPlaceholder"

describe("CurriculumListLoadingPlaceholder", () => {
  it("should render correctly", () => {
    const { container } = render(<CurriculumListLoadingPlaceholder />)
    expect(container).toMatchSnapshot()
  })
})
