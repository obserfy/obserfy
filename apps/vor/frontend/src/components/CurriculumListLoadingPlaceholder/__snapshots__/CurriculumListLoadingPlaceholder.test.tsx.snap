// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurriculumListLoadingPlaceholder should render correctly 1`] = `
@keyframes animation-0 {
  0% {
    opacity: 0.05;
  }

  50% {
    opacity: 0.2;
  }

  100% {
    opacity: 0.05;
  }
}

@keyframes animation-0 {
  0% {
    opacity: 0.05;
  }

  50% {
    opacity: 0.2;
  }

  100% {
    opacity: 0.05;
  }
}

@keyframes animation-0 {
  0% {
    opacity: 0.05;
  }

  50% {
    opacity: 0.2;
  }

  100% {
    opacity: 0.05;
  }
}

.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  font-size: 16px;
}

@media screen and (min-width: 40em) {
  .emotion-0 {
    font-size: 20px;
  }
}

.emotion-1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  border-bottom-width: 1px;
  border-bottom-style: solid;
  border-bottom-color: var(--theme-ui-colors-border);
}

.emotion-2 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin: 16px;
  border-radius: 6px;
  -webkit-animation: 4s infinite animation-0;
  animation: 4s infinite animation-0;
  background-color: #a5a5a5;
  height: 21px;
  width: 100px;
}

.emotion-4 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin: 16px;
  border-radius: 6px;
  -webkit-animation: 4s infinite animation-0;
  animation: 4s infinite animation-0;
  background-color: #a5a5a5;
  height: 21px;
  width: 200px;
}

.emotion-6 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin: 16px;
  border-radius: 6px;
  -webkit-animation: 4s infinite animation-0;
  animation: 4s infinite animation-0;
  background-color: #a5a5a5;
  height: 21px;
  width: 150px;
}

<div>
  <div
    class="emotion-0"
  >
    <div
      class="emotion-1"
    >
      <div
        class="emotion-2"
      />
    </div>
    <div
      class="emotion-1"
    >
      <div
        class="emotion-4"
      />
    </div>
    <div
      class="emotion-1"
    >
      <div
        class="emotion-6"
      />
    </div>
    <div
      class="emotion-1"
    >
      <div
        class="emotion-2"
      />
    </div>
    <div
      class="emotion-1"
    >
      <div
        class="emotion-4"
      />
    </div>
    <div
      class="emotion-1"
    >
      <div
        class="emotion-6"
      />
    </div>
  </div>
</div>
`;
